src/main/java/sirius/biz/jobs/batch/file/VirtualFileExtractionJob.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.jobs.batch.file;
+
+import sirius.biz.jobs.batch.BatchJob;
+import sirius.biz.process.ProcessContext;
+import sirius.biz.process.logs.ProcessLog;
+import sirius.biz.storage.layer3.VirtualFile;
+import sirius.biz.storage.layer3.VirtualFileSystem;
+import sirius.biz.util.UnzipHelper;
+import sirius.kernel.commons.Files;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.nls.NLS;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * Provides a job able to extract archives from the {@link VirtualFileSystem}. The following file extensions are supported: {@link UnzipHelper#getSupportedFileExtensions()}.
+ */
+public class VirtualFileExtractionJob extends BatchJob {
+
+    @Part
+    private static VirtualFileSystem vfs;
+
+    @Part
+    private static UnzipHelper unzipHelper;
+
+    VirtualFileExtractionJob(ProcessContext process) {
+        super(process);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        VirtualFile sourceFile = process.require(VirtualFileExtractionJobFactory.SOURCE_PARAMETER);
+        Optional<VirtualFile> destinationDirectory =
+                VirtualFileExtractionJobFactory.DESTINATION_PARAMETER.get(process.getContext());
+        boolean shouldOverwriteExisting =
+                process.require(VirtualFileExtractionJobFactory.OVERWRITE_EXISTING_FILES_PARAMETER);
+
+        // by default we'll use the files directory to extract to
+        final VirtualFile targetDirectory =
+                destinationDirectory.orElseGet(() -> vfs.resolve(sourceFile.parent().path()));
+
+        sourceFile.tryDownload().ifPresent(fileHandle -> {
+            File tempFile = fileHandle.getFile();
+            try {
+                UnzipHelper.unzip(tempFile,
+                                  null,
+                                  (status, data, filePath, filesProcessedSoFar, bytesProcessedSoFar, totalBytes) -> {
+                                      VirtualFile targetFile =
```

Comment:
this lambda is waaaay too long

Reply:
Lambda params come from `sirius.biz.util.ArchiveHelper.ArchiveExtractCallback#call`. Maybe an option would be to instantiate a new instance of an inner class instead of using this lambda?